crates/service/src/index/mod.rs

```diff
@@ -170,6 +172,25 @@ impl<S: G> Index<S> {
     pub fn view(&self) -> Arc<IndexView<S>> {
         self.view.load_full()
     }
+    pub fn setting(&self, key: String, value: String) -> Result<(), SettingError> {
+        match key.as_str() {
+            "optimizing.threads" => {
+                let parsed = usize::from_str(value.as_str())
+                    .map_err(|_e| SettingError::BadValue { key, value })?;
+                self.options
+                    .optimizing
+                    .optimizing_threads
+                    .store(parsed, Ordering::Relaxed);
+            }
+            _ => return Err(SettingError::BadKey { key }),
+        };
+        std::fs::write(
```

Comment:
Will an update lost or concurrent filesystem write happen if there are multiple `setting` calls at the same time?